.github/scripts/build.js

```diff
@@ -108,6 +122,47 @@ async function getCommandFromComment({ core, context, github }) {
   return command;
 }
 
+async function handleTestCommand({ core, github }, commandParts, testDescription, runId, pr, comment) {
+
+  if (!pr.details.mergeable) {
+    // Since we use the potential merge commit as the ref to checkout, we can only run if there is such a commit
+    // If the PR isn't mergeable, add a comment indicating that the merge issue needs addressing
+    const message = `:warning: Cannot run tests as PR is not mergeable. Ensure that the PR is open and doesn't have any conflicts.`;
+    await addActionComment({ github }, pr.repoOwner, pr.repoName, pr.number, comment.username, comment.link, message);
+    return false;
+  }
+
+  // check if this is an external PR (i.e. author not a maintainer)
+  // if so, need to specify the SHA that has been vetted and check that it matches
+  // the latest head SHA for the PR
+  const command = commandParts[0]
+  const prAuthorHasWriteAccess = await userHasWriteAccessToRepo({ core, github }, pr.authorUsername, pr.repoOwner, pr.repoName);
+  const externalPr = !prAuthorHasWriteAccess;
+  if (externalPr) {
+    if (commandParts.length === 1) {
+      const message = `:warning: When using \`${command}\` on external PRs, the SHA of the checked commit must be specified`;
+      await addActionComment({ github }, pr.repoOwner, pr.repoName, pr.number, comment.username, comment.link, message);
+      return false;
+    }
+    const commentSha = commandParts[1];
+    if (commentSha.length < 7) {
+      const message = `:warning: When specifying a commit SHA it must be at least 7 characters (received \`234567\`)`;
```

Comment:
is `234567` supposed to be the commentSha?

Reply:
Ooops - update to insert the correct value!